src/sdk/pynni/nni/compression/torch/pruning/structured_pruning.py

```diff
@@ -53,9 +74,16 @@ def calc_mask(self, sparsity, wrapper, wrapper_idx=None):
             mask_bias = None
         mask = {'weight_mask': mask_weight, 'bias_mask': mask_bias}
 
-        filters = weight.size(0)
-        num_prune = int(filters * sparsity)
-        if filters < 2 or num_prune < 1:
+        num_total = weight.size(0)
+        num_prune = int(num_total * sparsity)
+        if self.preserve_round > 1:
+            num_preserve = num_total - num_prune
+            num_preserve = int(math.ceil(num_preserve * 1. / self.preserve_round) * self.preserve_round)
+            if num_preserve > num_total:
+                num_preserve = int(math.floor(num_total * 1. / self.preserve_round) * self.preserve_round)
```

Comment:
this logic is a little strange, might be better to use `num_total` instead of `floor`.

Reply:
Thanks, fixed.